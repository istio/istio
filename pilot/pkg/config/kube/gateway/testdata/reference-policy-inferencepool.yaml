apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: simple
    hostname: "*.domain.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
---
apiVersion: inference.networking.x-k8s.io/v1alpha2
kind: InferencePool
metadata:
  name: my-ip
  namespace: inferencepool
spec:
  extensionRef:
    failureMode: FailOpen
    group: ""
    kind: Service
    name: endpoint-picker-svc
  selector:
    app: model-server
  targetPortNumber: 3000
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: ReferenceGrant
metadata:
  name: allow-service-ip
  namespace: inferencepool
spec:
  from:
  - group: gateway.networking.k8s.io
    kind: HTTPRoute
    namespace: istio-system
  to:
  - group: inference.networking.x-k8s.io
    kind: InferencePool
    name: my-ip
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: backend-allowed-ip
  namespace: istio-system
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["simple.domain.example"]
  rules:
  - backendRefs:
    - name: my-ip
      kind: InferencePool
      group: inference.networking.x-k8s.io
      namespace: inferencepool
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: backend-not-allowed-ip
  namespace: istio-system
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["simple2.domain.example"]
  rules:
  - backendRefs:
    - name: my-ip
      kind: InferencePool
      group: inference.networking.x-k8s.io
      namespace: inferencepool
      port: 80
      weight: 1
    - name: httpbin
      namespace: default
      port: 80
      weight: 1
