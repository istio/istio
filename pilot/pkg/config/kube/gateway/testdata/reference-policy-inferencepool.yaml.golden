apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/simple.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-simple
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*.domain.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/backend-allowed-ip.istio-system
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: istio-system~gateway-istio-autogenerated-k8s-gateway-simple~simple.domain.example
  namespace: istio-system
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-simple
  hosts:
  - simple.domain.example
  http:
  - name: istio-system.backend-allowed-ip.0
    route:
    - destination: {}
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/backend-not-allowed-ip.istio-system
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: istio-system~gateway-istio-autogenerated-k8s-gateway-simple~simple2.domain.example
  namespace: istio-system
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-simple
  hosts:
  - simple2.domain.example
  http:
  - name: istio-system.backend-not-allowed-ip.0
    route:
    - destination: {}
      weight: 1
    - destination: {}
      weight: 1
---
